pkg/transparentproxy/kubernetes/kubernetes.go

```diff
@@ -158,6 +157,22 @@ func NewPodRedirectForPod(pod *kube_core.Pod) (*PodRedirect, error) {
 	return podRedirect, nil
 }
 
+func excludeVirtualProbePort(annotations map[string]string) string {
+	// the annotations are validated/defaulted in a previous step in injector.NewAnnotations, so we can safely ignore the errors here
+	inboundPortsToExclude, _ := metadata.Annotations(annotations).GetString(metadata.KumaTrafficExcludeInboundPorts)
+	vpEnabled, _, _ := metadata.Annotations(annotations).GetEnabled(metadata.KumaVirtualProbesAnnotation)
+	if !vpEnabled {
+		return inboundPortsToExclude
+	}
+
+	vpPort, _ := metadata.Annotations(annotations).GetString(metadata.KumaVirtualProbesPortAnnotation)
+	if inboundPortsToExclude == "" {
+		return vpPort
+	}
+
+	return fmt.Sprintf("%s,%s", inboundPortsToExclude, vpPort)
+}
```

Comment:
I'm very uneasy with this... Why do we not have something similar to this with the current probes?

Reply:
Discussed in Slack.